dali/python/nvidia/dali/experimental/dynamic/_batch.py

```diff
@@ -446,12 +510,20 @@ def to_device(self, device: Device, force_copy: bool = False) -> "Batch":
                 return ret
 
     def cpu(self) -> "Batch":
+        """
+        Returns the batch on the CPU. If it's already there, this function returns `self`.
+        """
         return self.to_device(Device("cpu"))
 
     def gpu(self, index: Optional[int] = None) -> "Batch":
+        """
+        Returns the batch on the GPU. If it's already there, this function returns `self`.
+
+        If index is not specified, the current CUDA device is used.
+        """
         return self.to_device(Device("gpu", index))
 
-    def assign(self, other: "Batch"):
+    def _assign(self, other: "Batch"):
```

Comment:
I am on a fence if we maybe should also document internal methods for our benefit.